tests/test_samples.py

```diff
@@ -1517,47 +1517,47 @@ def test_groupby_stats():
         if 'chain' not in col:
             for chain in [1, 2]:
                 mask = mcmc.chain == chain
-                assert_allclose(mcmc.loc[mask, col].mean(),
+                assert_allclose(mcmc.loc[mask][col].mean(),
```

Comment:
I worry there is some underlying change here that needs to be understood

Reply:
I think the problem here is that `mask` is not a simple boolean array. Would it work to leave things as they are and only change the mask?
```python
mask = (mcmc.chain == chain).to_numpy()
```

Reply:
yes, but then I'm puzzled why the mask in the loop before is fine without `.to_numpy()`, even though it is also weighted

Reply:
Agreed, seems slightly inconsistent, but I think quite generally it is deemed cleaner to use a list/array as a boolean mask rather than a pandas series/dataframe which includes an index.

Reply:
I have to say I was surprised it worked at all

Reply:
I've changed the other mask to match too